src/mds/MDCache.cc

```diff
@@ -11200,8 +11206,9 @@ void MDCache::handle_dentry_unlink(const cref_t<MDentryUnlink> &m)
 {
   // straydn
   CDentry *straydn = nullptr;
+  CInode *strayin = nullptr;
   if (m->straybl.length())
-    decode_replica_stray(straydn, m->straybl, mds_rank_t(m->get_source().num()));
+    decode_replica_stray(straydn, &strayin, m->straybl, mds_rank_t(m->get_source().num()));
```

Comment:
@vshankar could you point me to the location where `strayin` is used after decoding

Reply:
MDCache::decode_replica_stray() will link the decoded inode (which would have stray_prior_path) and that would get checked in SessionMap::check_access().

Reply:
May I know what are these stray directories?

Reply:
Inodes that are deleted are _moved_ to special (per-mds) directory for purging at a later time point. The actual deletion is handled by PurgeQueue and StrayManager classes.